src/Microsoft.Health.Dicom.Core/Features/Query/QueryService.cs

```diff
@@ -168,11 +169,12 @@ private static bool ReturnNewTagDefaults(int? version)
             else
             {
                 Dictionary<string, StudyResult> map = studyComputedResults.ToDictionary<StudyResult, string>(a => a.StudyInstanceUid, StringComparer.OrdinalIgnoreCase);
-                instanceMetadata = instanceMetadata.Select(x =>
-                {
-                    var ds = new DicomDataset(x);
-                    return ds.AddOrUpdate(map[x.GetSingleValue<string>(DicomTag.StudyInstanceUID)].DicomDataset);
-                });
+                instanceMetadata = instanceMetadata.Where(x =>
```

Comment:
Instead of this, which does another loop of the list. can you do below on line 176?